Game.java

```diff
@@ -62,46 +59,34 @@ public void assignPlayers() {
         }
     }
 
-    public void playGame(Board board) {
+    public void playGame() {
         assignPlayers();
-        print(gameMode.toUpperCase() + " game is starting..");
-        print("Where would you like to go? (1, 2, 3, 4, 5, 6, 7, 8, 9)");
-        while (!board.checkForDraw() || !board.getState().equals("Game Over")) {
+        gameType.drawNewGame(this);
+        while (!board.isGameOver()){
             board = playerMakesMove();
         }
-        System.out.println(playerActive + " has won this game!");
-    }
-
-    public enum Gamemodes {
-        PVP,
-        PVC,
-        CVC
-    }
-
-    public void displayAllGamemodes() {
-        print("Hello, what gamemode would you like?");
-        for (Gamemodes gm : Gamemodes.values()) {
-            print(gm.toString());
-        }
+        gameType.gameIsOver(board, this);
     }
 
     public void pickGameMode() {
-        displayAllGamemodes();
+        gameType.displayAllGameModes();
         Scanner s = new Scanner(System.in);
         String input = s.nextLine();
         for (Gamemodes gm : Gamemodes.values()) {
-            if (input.equals(gm.toString())) {
+            if (input.equals(gm.toString().toLowerCase()) || input.equals(gm.toString())) {
                 gameMode = input.toLowerCase();
-                playGame(board);
+                playGame();
                 return;
             }
         }
-        print("You did not pick a valid gamemode.");
+        gameType.invalidGamemode();
+        pickGameMode();
     }
 
     public static void main(String[] args) {
         Board board = new Board("---------");
-        Game game = new Game(System.in, board);
+        ConsoleGame consoleGame = new ConsoleGame();
+        Game game = new Game(System.in, board, consoleGame);
```

Comment:
Why System.in ?  Why not InputStream ?  You should use the same input stream for constructing ConsoleGame and Game..